lib/ClangImporter/ImportEnumInfo.cpp

```diff
@@ -337,6 +343,12 @@ void EnumInfo::determineConstantNamePrefix(const clang::EnumDecl *decl) {
         checkPrefix = checkPrefix.drop_front();
     }
 
+    // If there's no decl name, just use the prefix we already created.
+    if (!decl->hasNameForLinkage()) {
+      constantNamePrefix = commonPrefix;
+      return;
```

Comment:
I don't like this—the type name contributes to the prefix computation for a reason. We should probably continue to the code below with `enumNameStr` containing the underlying typedef's name instead of the enum's lack-of-name.

Reply:
(Code structuring note: there might be a case for a "get C name of type" helper which encompasses checking the decl's name, checking its linkage name, and then this special check for an underlying typedef. There might even *be* such a helper already that you could add to—I'm not sure.)